### ISSUE TITLE
Execution Block ID uid://A001/X1590/X30aa Sgr_A_st_a_03_TM1

### ISSUE THREAD

**Issue by keflavich** [ACES-CMZ/reduction_ACES#242]
Sgr_A_st_a_03_TM1
[uid://A001/X1590/X30aa](https://almascience.org/aq/?result_view=observation&mous=uid://A001/X1590/X30aa)

* [x] Observations completed?
* [ ] Delivered?
* [ ] Downloaded? (specify where)
  * [ ] hipergator
  * [ ] hipergator pipeline run

* [ ] [Weblog](https://data.rc.ufl.edu/secure/adamginsburg/ACES/weblogs/humanreadable/Sgr_A_st_a_03_TM1/html/) unpacked
* [ ] [Weblog](https://data.rc.ufl.edu/secure/adamginsburg/ACES/weblogs/humanreadable/Sgr_A_st_a_03_TM1/html/) Quality Assessment?
* [ ] Imaging: Continuum
* [ ] Imaging: Lines

  * [ ] SPW25 H13CN
  * [ ] SPW27 H13CO+/SiO
  * [ ] SPW29 HCO+
  * [ ] SPW31 HNCO
  * [ ] SPW33 Cont 1
  * [ ] SPW35 Cont 2

**Comment by d-l-walker** [comment]
Calibration and imaging look great for this. 
- Very little continuum structure detected.
- Plenty of line emission, HNCO in particular looks great.

Spectral averaging due to size mitigation, so the cubes will need to be recleaned at native resolution.

I also noticed that the image size and cell size have been adjusted for size mitigation. AFAIK, this is not seen commonly in other cases of size mitigation with our data. I assume we will want to correct for this in the recleaning as and when it crops up @keflavich?

![Screenshot 2022-10-05 at 10 55 47](https://user-images.githubusercontent.com/13015266/194036184-c1f0b8fb-4caa-4ac0-942a-539bbc4f8e15.png)

**Comment by keflavich** [comment]
Yes, we'll want to fix this.  We don't want any size mitigation at all.

This one failed in the pipeline restore run with a very weird problem:
```
2022-10-07 10:40:28 INFO: Setting plot level to 'default'
Traceback (most recent call last):
  File "/blue/adamginsburg/adamginsburg/casa/casa-6.4.3-2-pipeline-2021.3.0.17/lib/py/lib/python3.6/site-packages/casashell/private/init_system.py", line 238, in __evprop__
    exec(stmt)
  File "<string>", line 1, in <module>
  File "/blue/adamginsburg/adamginsburg/casa/casa-6.4.3-2-pipeline-2021.3.0.17/lib/py/lib/python3.6/site-packages/casashell/private/init_system.py", line 175, in execfile
    newglob = run_path( filename, init_globals=globals )
  File "/blue/adamginsburg/adamginsburg/casa/casa-6.4.3-2-pipeline-2021.3.0.17/lib/py/lib/python3.6/runpy.py", line 263, in run_path
    pkg_name=pkg_name, script_name=fname)
  File "/blue/adamginsburg/adamginsburg/casa/casa-6.4.3-2-pipeline-2021.3.0.17/lib/py/lib/python3.6/runpy.py", line 96, in _run_module_code
    mod_name, mod_spec, pkg_name, script_name)
  File "/blue/adamginsburg/adamginsburg/casa/casa-6.4.3-2-pipeline-2021.3.0.17/lib/py/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "imaging_pipeline_rerun.py", line 25, in <module>
    raise ValueError("No MSes found")
ValueError: No MSes found
```
It didn't even create the `calibrated/` directory!

This is super weird.  I assume it's caused by the ASDMs being absent... I'll see if that's true.

**Comment by keflavich** [comment]
@piposona could you have a look at this one too?  Basically, does the pipeline work when you run it?

**Comment by piposona** [comment]
Downloading. @keflavich , can you confirm the following files are in globus? I cannot find them, and they are required for the restoring of this one.
uid___A002_Xfdf45f_X4f4.asdm.sdm.tar
uid___A002_Xfdf45f_X38ff.asdm.sdm.tar

**Comment by keflavich** [comment]
Well if they're missing, that's the problem.  But I see:
```
-rw-r--r-- 1 adamginsburg adamginsburg 55G Oct  4 14:35 /orange/adamginsburg/ACES/data/2021.1.00172.L_uid___A002_Xfdf45f_X38ff.asdm.sdm.tar
-rw-r--r-- 1 adamginsburg adamginsburg 53G Oct  4 18:31 /orange/adamginsburg/ACES/data/2021.1.00172.L_uid___A002_Xfdf45f_X4f4.asdm.sdm.tar
```
they follow a different naming convention (they have the program ID at the front), but they're there.

**Comment by keflavich** [comment]
It is possible this actually did restore correctly when I re-re-ran the pipeline:
```
drwxr-sr-x 28 adamginsburg adamginsburg 4.0K Oct  7 13:25 2021.1.00172.L/science_goal.uid___A001_X1590_X30a8/group.uid___A001_X1590_X30a9/member.uid___A001_X1590_X30aa/calibrated/working/uid___A002_Xfdf45f_X4f4.ms
drwxr-sr-x 28 adamginsburg adamginsburg 4.0K Oct  7 14:01 2021.1.00172.L/science_goal.uid___A001_X1590_X30a8/group.uid___A001_X1590_X30a9/member.uid___A001_X1590_X30aa/calibrated/working/uid___A002_Xfdf45f_X38ff.ms
drwxr-sr-x 28 adamginsburg adamginsburg 4.0K Oct  8 02:46 2021.1.00172.L/science_goal.uid___A001_X1590_X30a8/group.uid___A001_X1590_X30a9/member.uid___A001_X1590_X30aa/calibrated/working/uid___A002_Xfdf45f_X38ff_target.ms
drwxr-sr-x 28 adamginsburg adamginsburg 4.0K Oct  8 02:52 2021.1.00172.L/science_goal.uid___A001_X1590_X30a8/group.uid___A001_X1590_X30a9/member.uid___A001_X1590_X30aa/calibrated/working/uid___A002_Xfdf45f_X4f4_target.ms
```
To be sure, we have to check the images and do some more checks on the MSes.  It looks like the pipeline did eventually core dump:
```
2021.1.00172.L/science_goal.uid___A001_X1590_X30a8/group.uid___A001_X1590_X30a9/member.uid___A001_X1590_X30aa/calibrated/working/core.python3-6-1665366649-3663-3389-c0711a-s2.ufhpc.36226
2021.1.00172.L/science_goal.uid___A001_X1590_X30a8/group.uid___A001_X1590_X30a9/member.uid___A001_X1590_X30aa/calibrated/working/core.python3-6-1665366648-3663-3389-c0711a-s2.ufhpc.36244
```
but based on the `IMAGING_WEIGHT` and `TempLattice` debris it left behind, I think that happened during the late imaging stage and is not a concern.

**Comment by piposona** [comment]
> they follow a different naming convention (they have the program ID at the front), but they're there.

Totally right, i missed that!

**Comment by keflavich** [comment]
Unfortunately I think this makes it a more severe problem.  We can't use 6.2.1-7 because of an error - my note in the pipeline runner is "# 6.2.1 is the 2021 release version, but it contains the readBlock error". I'll have to look up what that means... _maybe_ I can regress to 6.2.1.7, but not definitely.

**Comment by piposona** [comment]
Ok, let me know how to proceed. Happy to process those giving problems, rather than spending time figuring out the issues with CASA versions.

**Comment by keflavich** [comment]
yeah, go ahead and upload the MS - at least we can get cubes.  But we still need to get to the bottom of this _eventually_

**Comment by piposona** [comment]
@keflavich , I have just uploaded the .ms and the .msflagversion to the following directory
``
/upload/member.uid___A001_X1590_X30aa/
``
Let me know if you want to try to get the root cause of this issue. I will not delete the data for the time being.

**Comment by keflavich** [comment]
Thanks!  Do you also have the `_target.ms` split?

**Comment by piposona** [comment]
For some reason this is not generated. There is a parameter in the scriptForPI which is by default
``
dosplit = False
``
I could run it again with the parameter to True

**Comment by keflavich** [comment]
hmmm... it shouldn't _just_ be caused by dosplit, b/c the `_target.ms` includes the continuum subtracted data; did your run do the cont sub?

**Comment by piposona** [comment]
Just run the script as is...that is
``
docontsub = False
``
this can be edited or overridden with global parameters, but just run it as is.
Happy to repeat as needed.

**Comment by keflavich** [comment]
that's weird - I see that `docontsub` is set to `False` as you say, but it produces the split `_target.ms` for every other field.  I don't think I've specified any kind of override.

**Comment by piposona** [comment]
This is indeed weird. During the investigation of issue #145, we found a similar difference between the person running the script in the data processing cluster and my running it in a different machine. And apparently there were no global variables overriding the script defaults. 
I will further investigate with the person in charge of the #145 investigation

**Comment by keflavich** [comment]
The issues for this field appear to be resolved; the rerun mentioned in https://github.com/ACES-CMZ/reduction_ACES/issues/242#issuecomment-1276255117 was successful and images have been produced from the reimaging scripts:
```
drwxr-sr-x 4 adamginsburg adamginsburg 4.0K Oct 14 05:21 2021.1.00172.L/science_goal.uid___A001_X1590_X30a8/group.uid___A001_X1590_X30a9/member.uid___A001_X1590_X30aa/calibrated/working/uid___A001_X1590_X30aa.s38_0.Sgr_A_star_sci.spw27.cube.I.iter1.image
drwxr-sr-x 4 adamginsburg adamginsburg 4.0K Oct 15 14:34 2021.1.00172.L/science_goal.uid___A001_X1590_X30a8/group.uid___A001_X1590_X30a9/member.uid___A001_X1590_X30aa/calibrated/working/uid___A001_X1590_X30aa.s38_0.Sgr_A_star_sci.spw25.cube.I.iter1.image
drwxr-sr-x 4 adamginsburg adamginsburg 4.0K Oct 16 05:41 2021.1.00172.L/science_goal.uid___A001_X1590_X30a8/group.uid___A001_X1590_X30a9/member.uid___A001_X1590_X30aa/calibrated/working/uid___A001_X1590_X30aa.s38_0.Sgr_A_star_sci.spw35.cube.I.iter1.image
drwxr-sr-x 4 adamginsburg adamginsburg 4.0K Oct 16 15:25 2021.1.00172.L/science_goal.uid___A001_X1590_X30a8/group.uid___A001_X1590_X30a9/member.uid___A001_X1590_X30aa/calibrated/working/uid___A001_X1590_X30aa.s38_0.Sgr_A_star_sci.spw31.cube.I.iter1.image
```

Let's keep an eye on this one though.  

It's the funniest-looking footprint you've ever seen, but it's pretty:
![image](https://user-images.githubusercontent.com/143715/196190852-f447f5c6-b838-4dbf-81aa-43b1d444cc65.png)

**Comment by piposona** [comment]
Ok, I see. I had not realized that you had finally made it work and the data had been properly restored. Somehow I had totally ignored this comment https://github.com/ACES-CMZ/reduction_ACES/issues/242#issuecomment-1276255117
So I assume that the data that I uploaded were not needed. I was now producing the splitted versions.

Let me know if any action is needed on my side.

**Comment by d-l-walker** [comment]
@keflavich I've just updated the tclean parameters in PR #320 to undo the size mitigation for this region. Apologies for not doing this sooner -- I must have lost track of this due to all of the earlier pipeline issues.

**Comment by d-l-walker** [comment]
@keflavich I forgot to mention in my previous comment -- could you move the size mitigated products, and re-run the cleaning for this SB with the updated parameters? Thanks!

**Comment by keflavich** [comment]
done; moved everything away and (attempted) reimaging

**Comment by keflavich** [comment]
Still running.

**Comment by d-l-walker** [comment]
This appears to have finished reprocessing all SPWs now. I'll check the products and update the issue.

**Comment by d-l-walker** [comment]
QA for latest continuum selection/imaging:
* Not a whole lot going on in this region
* Some very faint extended emission in the newer low/high frequency SPW images
* Nothing obvious in the individual MFS images
* New SPW 33 & 35 image shows a handful of compact sources much more clearly than before

I think this one is OK. But I'll grab the SPW 25 & 29 cubes just to double check.

![region_a_continuum](https://github.com/ACES-CMZ/reduction_ACES/assets/13015266/08b16d8c-4947-47a9-93b6-a9775b080734)

![region_a_continuum_crop](https://github.com/ACES-CMZ/reduction_ACES/assets/13015266/9b5b9ffd-ab9e-4560-9afb-4098d8f923b6)

**Comment by d-l-walker** [comment]
I think this is the culprit for the faint line contamination in SPW 27, which is then present in the spw25_27 continuum image.

Left: SPW 27 cube, Right: SPW 27 mfs
![Screenshot 2024-03-06 at 10 59 24](https://github.com/ACES-CMZ/reduction_ACES/assets/13015266/9e0637b6-da48-48f9-85d3-7895ce20d3bd)

**Comment by keflavich** [comment]
OK, I see a problem: the maxspec shown in this figure:
![image](https://github.com/ACES-CMZ/reduction_ACES/assets/143715/c5e3e17d-1e9c-4f90-a87b-522090be2d43)

does not have the lines shown in spw27, so it must have been calculated incorrectly.

**Comment by keflavich** [comment]
This issue helped me ID a problem.  Both of these are max spectra of spw 27, but the first is pbcor (and contsub), the second is non-pbcor

![image](https://github.com/ACES-CMZ/reduction_ACES/assets/143715/a8caede2-96bd-47db-a76f-ced7962a5d88)

![image](https://github.com/ACES-CMZ/reduction_ACES/assets/143715/f3332364-0ee6-4513-9de2-3dc7716de6bb)

**Comment by d-l-walker** [comment]
New continuum images look better. In particular the line contamination in the low freq images has clearly been resolved. Very little continuum structure in this region.

Left: old images
Right: new images
![Screenshot 2024-03-27 at 12 02 46](https://github.com/ACES-CMZ/reduction_ACES/assets/13015266/ca3bcaa2-68b2-466d-a2af-2b92ac1e6e88)
